src/java/eu/rssw/rcode/ClassCompilationUnit.java

```diff
@@ -0,0 +1,217 @@
+            boolean found = false;
+            for (Method m : methods) {
+                if (m.methodName.equalsIgnoreCase(m2.methodName)
+                        && hasSameSignature(m.parameters, m2.parameters)) {
+                    m.methodComment = m2.methodComment;
+                    found = true;
+                }
+            }
+            if (!found) {
+                System.out.println("not found");
+                methods.add(m2);
+            }
+        }
+
+        // Fusion des using
+        usings.addAll(unit.usings);
+    }
+
+    /**
+     * Verify signature.
+     * 
+     * @param params1 RCode signature
+     * @param params2 Parser signature
+     * @return True if same signature
+     */
+    private static boolean hasSameSignature(List<Parameter> params1, List<Parameter> params2) {
+        if (params1.size() != params2.size())
+            return false;
+        for (Parameter p1 : params1) {
+            for (Parameter p2 : params2) {
+                if (p1.mode != p2.mode)
+                    return false;
+                if (!p1.dataType.endsWith(p2.dataType))
+                    return false;
+            }
+        }
+        return true;
+    }
+
+    public static ClassCompilationUnit readClassInfo(RCodeClassInfo classInfo)
+            throws IOException, RCodeReadException {
+        ClassCompilationUnit unit = new ClassCompilationUnit();
+        if (classInfo.getName().lastIndexOf('.') == -1) {
+            unit.packageName = "";
+            unit.className = classInfo.getName();
+        } else {
+            unit.className = classInfo.getName()
+                    .substring(classInfo.getName().lastIndexOf('.') + 1);
+            unit.packageName = classInfo.getName().substring(0,
+                    classInfo.getName().lastIndexOf('.'));
+        }
+        unit.inherits = classInfo.getBaseType();
+        unit.isAbstract = classInfo.isAbstract();
+        unit.isFinal = classInfo.isFinal();
+        unit.isInterface = classInfo.isInterface();
+
+        for (TypeInformation inf : classInfo.getInterfaces()) {
+            unit.interfaces.add(inf.getName());
+        }
+        for (PropertyElement elem : classInfo.getProperties()) {
+            unit.properties.add(Property.readPropertyElement(elem));
+        }
+        for (EventElement elem : classInfo.getEvents()) {
+            unit.events.add(Event.readEventElement(elem));
+        }
+        for (MethodElement elem : classInfo.getConstructors()) {
+            unit.constructors.add(Constructor.readMethodElement(elem));
+        }
+        for (MethodElement elem : classInfo.getMethods()) {
+            Method m = Method.readMethodElement(elem);
+            if (m != null)
+                unit.methods.add(m);
+        }
+        return unit;
+    }
+
+    @SuppressWarnings("unused")
+    private static Comment parseComments(String comments) {
+        Comment comm = new Comment();
+        comments = comments.trim();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "comments". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1226)